docs/data/sdks/go.md

```diff
@@ -0,0 +1,405 @@
+Revenue (optional) | float64 | Use negative values to indicate refunds. Note: Revenue = Quantity * Price | 0
+
+### `Flush`
+
+The `Flush` method triggers the client to send buffered events.
+
+```Go
+client.Flush()
+```
+
+### `Add`
+
+The `Add` method adds a plugin to Amplitude client struct. Plugins can help processing and sending events. [Learn more about plugins.](#amplitude-sdk-plugin).
+
+```Go
+client.add(pluginObj)
+```
+
+### `Remove`
+
+The `Remove` method removes the given plugin from the client struct if exists.
+
+```Go
+client.Remove(pluginObj)
+```
+
+### Shutdown
+
+`Shutdown` method closes the client struct. Closed client struct will not accepting new events and will try to flush events left in buffer. Then the client struct will shutdown running threads.
+
+```Go
+client.Shutdown()
+```
+
+## Amplitude SDK Plugin
+
+Plugins allow you to extend Amplitude SDK's behavior by, for example, modifying event properties (enrichment type) or sending to a third-party APIs (destination type). A plugin is struct with methods `Setup()` and `Execute()`.
+
+### Plugin.Setup
+
+This method contains logic for preparing the plugin for use and has `Config` struct as a parameter. The expected return value is `nil`. A typical use for this method, is to copy configuration from `Config` or instantiate plugin dependencies. This method is called when the plugin is registered to the client via `client.Add()`.
+
+### Plugin.Execute
+
+This method contains the logic for processing events and has `*Event` as parameter. If used as enrichment type plugin, the expected return value is the modified/enriched event; while if used as a destination type plugin, the expected return value is `nil`. This method is called for each event, including Identify, GroupIdentify and Revenue events, that is instrumented using the client interface.
+
+### Plugin Examples
+
+#### Enrichment Type Plugin
+
+Here's an example of a plugin that modifies each event that is instrumented by adding an increment integer to `EventId` property of an event.
+
+```Go
+type addEventPlugin struct {
+	currentID int
+	config    amplitude.Config
+}
+
+func (plugin *addEventPlugin) Setup(config amplitude.Config) {
+	plugin.config = config
+}
+
+func (plugin *addEventPlugin) Priority() amplitude.EnrichmentPriority {
+	return amplitude.EnrichmentPriorityEnrichment
+}
+
+func (plugin *addEventPlugin) Execute(event *amplitude.Event) *amplitude.Event {
+	event.EventID = plugin.currentID
+	plugin.currentID += 1
+	return event
+}
+  
+config := amplitude.NewConfig("your-api-key")
+client := amplitude.NewClient(config)
+client.Add(&addEventPlugin{})
+```
+
+#### Destination Type Plugin
+
+```Go
+type myDestinationPlugin struct{
+	url string
+	config amplitude.Config
+}
+
+func (plugin *myDestinationPlugin) Setup(config amplitude.Config) {
+	plugin.config = config
+}
+
+func (plugin *myDestinationPlugin) Execute(event *amplitude.Event) {
+	payload := map[string]interface{}{"key": "secret", "event":event}
```

Comment:
```suggestion
	payload := map[string]interface{}{"key": "secret", "events":event}
```

Reply:
Also events payload should be an array

Reply:
It should be just a single event. Typescript example https://www.docs.developers.amplitude.com/data/sdks/typescript-node/#destination-type-plugin